File: test_archives/py_tests/batch5/test_function5.py
from functions.function5 import function5
import pytest

@pytest.fixture
def test_class():
    test_class = function5('hello', 5, 3.4, True)
    return test_class

def test_process_data_1(test_class):
    assert test_class.process_data() == 56.0

def test_process_data_2(test_class):
    test_class.input_string = 'world'
    assert test_class.process_data() == 56.0

File: test_archives/cpp_tests/batch5/test_function5.cpp
#include "../catch.hpp"
#include "../functions/function5.cpp"

TEST_CASE("test_process_data_1") {
    function5 test_class("hello", 5, 3.4, true);
    REQUIRE(test_class.process_data() == 56.0);
}

TEST_CASE("test_process_data_2") {
    function5 test_class("hello", 5, 3.4, true);
    test_class.input_string = "world";
    REQUIRE(test_class.process_data() == 56.0);
}